### ISSUE TITLE
Certain items should not be consumed when used

### ISSUE THREAD

**Issue by carpenterd777** [StevensSEC/pokemonbattlelib#104]
We'll need some sort of mechanism to prevent certain items from being consumed when they incur an `ItemTransaction`, an example would be the Quick Claw. This may be possible to implement based on the `Category` field of `Item`.

**Comment by dyc3** [comment]
I don't think that quick claw is a great example. It's just a stat boost item isn't it? Doesn't the stat boost get removed if the item is removed during the battle?

**Comment by carpenterd777** [comment]
No, quick claw does not provide a stat boost, it  has a 20% chance to activate and when it does the Pokemon holding it moves first in its priority bracket.
see: https://bulbapedia.bulbagarden.net/wiki/Quick_Claw

**Comment by dyc3** [comment]
Ah, you're right

**Comment by dyc3** [comment]
We should give items flags for properties they have.
